appvia/kore
# Fix AKS nodegroup handling + validation/autoscaling improvements

## Summary

Fix AKS cluster nodegroup handling, as nodegroups has to be managed using a different API endpoint.

Also the PR adds extra cluster validation:
 - all node groups must have a unique name in a cluster
 - in an AKS cluster at least one node group must be a System one

Also I had to do a workaround when increasing the minimum node count if it would be higher than the current node count:
 - disable autoscaling temporarily and scale the node group up to the minimum node count
 - re-enable autoscaling

**Which issue(s) this PR resolves**:

Resolves #1124

## Testing

See basic setup instructions here: https://github.com/appvia/kore/pull/1048, if you've never tested Azure before.

 * Create an AKS cluster
 * Add a new node group
 * Update an existing node group (e.g. change the minimum node count)
 * Delete an existing node group

## Follow-up stories

The immutable fields are not checked on node groups: https://github.com/appvia/kore/issues/1130
